init-tools.sh

```diff
@@ -12,126 +12,116 @@ __BUILD_TOOLS_PACKAGE_VERSION=$(cat $__scriptpath/BuildToolsVersion.txt)
 __DOTNET_TOOLS_VERSION=$(cat $__scriptpath/DotnetCLIVersion.txt)
 __BUILD_TOOLS_PATH=$__PACKAGES_DIR/microsoft.dotnet.buildtools/$__BUILD_TOOLS_PACKAGE_VERSION/lib
 __INIT_TOOLS_RESTORE_PROJECT=$__scriptpath/init-tools.msbuild
-__INIT_TOOLS_DONE_MARKER_DIR=$__TOOLRUNTIME_DIR/$__BUILD_TOOLS_PACKAGE_VERSION
-__INIT_TOOLS_DONE_MARKER=$__INIT_TOOLS_DONE_MARKER_DIR/done
+__BUILD_TOOLS_SEMAPHORE=$__TOOLRUNTIME_DIR/$__BUILD_TOOLS_PACKAGE_VERSION/init-tools.complete
 
-# We do not want to run the first-time experience.
-export DOTNET_SKIP_FIRST_TIME_EXPERIENCE=1
+if [ -e $__BUILD_TOOLS_SEMAPHORE ]; then
+    echo "Tools are already initialized"
+    return #return instead of exit because this script is inlined in other scripts which we don't want to exit
+fi
+
+if [ -e $__TOOLRUNTIME_DIR ]; then rm -rf -- $__TOOLRUNTIME_DIR; fi
 
-if [ -z "$__DOTNET_PKG" ]; then
-    if [ "$(uname -m | grep "i[3456]86")" = "i686" ]; then
-        echo "Warning: build not supported on 32 bit Unix"
+if [ -d "$DotNetBuildToolsDir" ]; then
+    echo "Using tools from '$DotNetBuildToolsDir'."
+    ln -s "$DotNetBuildToolsDir" "$__TOOLRUNTIME_DIR"
+
+    if [ ! -e "$__DOTNET_CMD" ]; then
+        echo "ERROR: Ensure that $DotNetBuildToolsDir contains the .NET Core SDK at $__DOTNET_PATH"
+        exit 1
     fi
-OSName=$(uname -s)
-    case $OSName in
-        Darwin)
-            OS=OSX
-            __DOTNET_PKG=dotnet-dev-osx-x64
-            ulimit -n 2048
-            ;;
-
-        Linux)
-            __DOTNET_PKG=dotnet-dev-linux-x64
-            OS=Linux
-
-            if [ -e /etc/os-release ]; then
-                source /etc/os-release
-                if [[ $ID == "alpine" ]]; then
-                    # remove the last version digit
-                    VERSION_ID=${VERSION_ID%.*}
-                    __DOTNET_PKG=dotnet-dev-alpine.$VERSION_ID-x64
-                fi
-            elif [ -e /etc/redhat-release ]; then
-                redhatRelease=$(</etc/redhat-release)
-                if [[ $redhatRelease == "CentOS release 6."* || $redhatRelease == "Red Hat Enterprise Linux Server release 6."* ]]; then
-                    __DOTNET_PKG=dotnet-dev-rhel.6-x64
-                fi
-            fi			
-			
-            ;;
-
-        *)
-            echo "Unsupported OS '$OSName' detected. Downloading linux-x64 tools."
-            OS=Linux
-            __DOTNET_PKG=dotnet-dev-linux-x64
-            ;;
-  esac
+
+    echo "Done initializing tools."
+    mkdir -p "$(dirname "$__BUILD_TOOLS_SEMAPHORE")" && touch $__BUILD_TOOLS_SEMAPHORE
+    exit 0
```

Comment:
You can do this as a follow-up but I've sense changed this to be a return 0 as well. It is really only interesting in the source-build scenario though.